nipype/algorithms/compcor.py

```diff
@@ -0,0 +1,81 @@
+from ..interfaces.base import (BaseInterfaceInputSpec, TraitedSpec,
+                               BaseInterface, traits, File)
+import nibabel as nb
+import numpy as np
+from scipy import linalg, stats
+import os
+
+from nipype.pipeline.engine import Workflow
+
+class CompCoreInputSpec(BaseInterfaceInputSpec):
+    realigned_file = File(exists=True, mandatory=True, desc='already realigned brain image (4D)')
+    mask_file = File(exists=True, mandatory=True, desc='mask file that determines ROI (3D)')
+    num_components = traits.Int(6, usedefault=True) # 6 for BOLD, 4 for ASL
+    # additional_regressors??
+
+class CompCoreOutputSpec(TraitedSpec):
+    components_file = File(desc='text file containing the noise components', exists=True)
+
+class CompCore(BaseInterface):
```

Comment:
Not the best place to mention this (I can make a separate issue) but is there a keyword search functionality in nipype? In other words it would be great if interfaces had keywords (like "motion correction", "brain masking"...it would be important to choose them carefully!) and then there was some way to search through them to find an interface that does what you want. Seems relevant here as it is best to have the name match what's in the paper, but then that name isn't actually all that descriptive about what the interface does!